code/datums/components/overlay_lighting.dm

```diff
@@ -174,9 +208,13 @@
 	if(.)
 		var/atom/movable/old_parent_attached_to = .
 		UnregisterSignal(old_parent_attached_to, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+		if(directional)
+			UnregisterSignal(old_parent_attached_to, COMSIG_ATOM_DIR_CHANGE)
 		if(old_parent_attached_to == current_holder)
 			RegisterSignal(old_parent_attached_to, COMSIG_PARENT_QDELETING, .proc/on_holder_qdel)
 			RegisterSignal(old_parent_attached_to, COMSIG_MOVABLE_MOVED, .proc/on_holder_moved)
+			if(directional)
+				RegisterSignal(old_parent_attached_to, COMSIG_ATOM_DIR_CHANGE, .proc/on_holder_dir_change)
 	if(parent_attached_to)
 		if(parent_attached_to == current_holder)
 			UnregisterSignal(current_holder, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
```

Comment:
```suggestion
	if(parent_attached_to)
		if(parent_attached_to == current_holder)
			UnregisterSignal(current_holder, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
		if(directional)
			RegisterSignal(parent_attached_to, COMSIG_ATOM_DIR_CHANGE, .proc/on_holder_dir_change)
```
This is runtiming for not handling the parent-attached-to logic properly.
This is so that if the light is on a flashlight that is inserted on a gun (yea, I know, hacky), then the component tracks the gun instead of the flashlight.